service/models.py

```diff
@@ -0,0 +1,122 @@
+"""
+Models for Promotion Service
+
+All of the models are stored in this module
+
+Models
+------
+Promotion - A Promotion used in the eCommerce website
+- id: (int) primary key
+- title: (str) the name of the promotion
+- description: (str) string/text
+- promo_code: (str) the promo_code associated with this promotion 
+- promo_type: (str or enum) [BOGO | DISCOUNT | FIXED]
+- amount: (int) the amount of the promotion base on promo_type
+- start_date: (date) the starting date
+- end_date: (date) the ending date
+-----------
+promotion_products - The relationship between promotion and product
```

Comment:
Question: so for a site wide promotion, would we have to have a promotion_product entry for every product? Maybe we should instead have a field on the promotion table indicating if it is site wide? @liondon

Reply:
oh, yeah, you're right. I thought about this for one second but then forgot it :P